### PR TITLE
ocs-to-ocs: verify consumer is onboarded before offboarding

### PR DESCRIPTION
**Pull request by iamniting**
skip offboarding request if consumer is not onboarded.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Review by nb-ohad — APPROVED** [review]

File: controllers/storagecluster/external_resources.go [hunk 1]
```diff
@@ -285,6 +285,13 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
 
 	if isExternalOCSProvider(instance) {
+
+		// make sure consumer is onboarded before offboarding
+		if instance.Status.ExternalStorage.ConsumerID == "" {
+			r.Log.Info("Consumer is not onboarded, skipping a offboarding request.")
```
Comment (sp98) [inline controllers/storagecluster/external_resources.go:291]:
```suggestion
			r.Log.Info("Consumer is not onboarded. Skipping the offboarding request.")
```

File: controllers/storagecluster/external_resources.go [hunk 2]
```diff
@@ -285,6 +285,13 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
 
 	if isExternalOCSProvider(instance) {
+
+		// make sure consumer is onboarded before offboarding
```
Comment (sp98) [inline controllers/storagecluster/external_resources.go:289]: Comment can be a bit confusing.
```suggestion
		// skip offboarding if consumer is not on onboarded
```

**Review by sp98 — CHANGES_REQUESTED** [review]
nit

**Review by sp98 — APPROVED** [review]

**Review by kesavanvt — APPROVED** [review]

**Pull request merged**